tests/ModelTest.php

```diff
@@ -1073,7 +1088,13 @@ public function testCreateOrFirst()
         $this->assertFalse($user2->wasRecentlyCreated);
         $this->assertEquals($createdAt, $user1->created_at->getTimestamp());
         $this->assertEquals($createdAt, $user1->updated_at->getTimestamp());
+        if ($transaction) {
+            $this->assertEquals([], $events);
+        } else {
+            $this->assertEquals(['saving', 'creating'], $events);
+        }
```

Comment:
Why would there be no events when executing within a transaction? Considering both code paths:

 * When we're in a transaction, `User::createOrFirst` will delegate to `firstOrCreate` and return `(clone $this)->where($attributes)->first()`.
 * When we're not in a transaction, `createOrFirst()` will return `$this->create(array_merge($attributes, $values))`. I assume that always dispatches these two events regardless of whether the document exists server-side.

Is there any concern with these event expectations differing from Eloquent?

Either way, I think a comment explaining different assertions would be helpful.